precompiles/proxy/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+			.try_into()
+			.map_err(|_| revert("failed decoding proxy_type"))?;
+		let delay = input.read::<u32>()?.into();
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::add_proxy {
+			delegate: Runtime::AddressMapping::into_account_id(delegate),
+			proxy_type,
+			delay,
+		}
+		.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Unregister a proxy account for the sender.
+	/// The dispatch origin for this call must be Signed.
+	///
+	/// Parameters:
+	/// * delegate: The account that the caller would like to remove as a proxy.
+	/// * proxy_type: The permissions currently enabled for the removed proxy account.
+	/// * delay: The announcement period required of the initial proxy. Will generally be zero.
+	fn remove_proxy(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let mut input = handle.read_input()?;
+		input.expect_arguments(3)?;
+
+		let delegate: H160 = input.read::<Address>()?.into();
+		let proxy_type = input
+			.read::<u8>()?
+			.try_into()
+			.map_err(|_| revert("failed decoding proxy_type"))?;
+		let delay = input.read::<u32>()?.into();
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::remove_proxy {
+			delegate: Runtime::AddressMapping::into_account_id(delegate),
+			proxy_type,
+			delay,
+		}
+		.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Unregister all proxy accounts for the sender.
+	/// The dispatch origin for this call must be Signed.
+	/// WARNING: This may be called on accounts created by anonymous, however if done, then the
+	/// unreserved fees will be inaccessible. All access to this account will be lost.
+	fn remove_proxies(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+		let call = ProxyCall::<Runtime>::remove_proxies {}.into();
+
+		<RuntimeHelper<Runtime>>::try_dispatch(handle, Some(origin).into(), call)?;
+
+		Ok(succeed([]))
+	}
+
+	/// Checks if the caller has an account proxied with a given proxy type
+	///
+	/// Parameters:
+	/// * delegate: The account that the caller has maybe proxied
+	/// * proxyType: The permissions allowed for the proxy
+	fn is_proxy(handle: &mut impl PrecompileHandle) -> EvmResult<PrecompileOutput> {
+		let mut input = handle.read_input()?;
+		input.expect_arguments(2)?;
+
+		let real: H160 = input.read::<Address>()?.into();
+		let real = Runtime::AddressMapping::into_account_id(real);
+		let proxy_type: Runtime::ProxyType = input
+			.read::<u8>()?
+			.try_into()
+			.map_err(|_| revert("failed decoding proxy_type"))?;
+
+		let origin = Runtime::AddressMapping::into_account_id(handle.context().caller);
+
+		let is_proxy = ProxyPallet::<Runtime>::proxies(origin)
```

Comment:
You need to add a db read cost here

` handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
`